src/core/utils/data-transform.ts

```diff
@@ -255,90 +260,50 @@ export function displace(
     }
 
     if (method === 'pile') {
-        const oldAlgorithm = false;
-
-        if (oldAlgorithm) {
-            // This will be deprecated soon.
-            const { maxRows } = t;
-            const boundingBoxes: { start: number; end: number; row: number }[] = [];
-
-            base.sort((a: Datum, b: Datum) => (a[startField] as number) - (b[startField] as number)).forEach(
-                (d: Datum) => {
-                    const start = (d[startField] as number) - padding;
-                    const end = (d[endField] as number) + padding;
-
-                    const overlapped = boundingBoxes.filter(
-                        box =>
-                            (box.start === start && end === box.end) ||
-                            (box.start <= start && start < box.end) ||
-                            (box.start < end && end <= box.end) ||
-                            (start < box.start && box.end < end)
-                    );
+        // This piling algorithm is heavily based on
+        // https://github.com/higlass/higlass-pileup/blob/8538a34c6d884c28455d6178377ee1ea2c2c90ae/src/bam-fetcher-worker.js#L626
+        const { maxRows } = t;
+        const occupiedSpaceInRows: { [group: string]: { start: number; end: number }[] } = {};
 
-                    // find the lowest non overlapped row
-                    const uniqueRows = [
-                        ...Array.from(new Set(boundingBoxes.map(d => d.row))),
-                        Math.max(...boundingBoxes.map(d => d.row)) + 1
-                    ];
-                    const overlappedRows = overlapped.map(d => d.row);
-                    const lowestNonOverlappedRow = Math.min(
-                        ...uniqueRows.filter(d => overlappedRows.indexOf(d) === -1)
-                    );
+        const sorted = base.sort((a: Datum, b: Datum) => (a[startField] as number) - (b[startField] as number));
 
-                    // row index starts from zero
-                    const row: number = overlapped.length === 0 ? 0 : lowestNonOverlappedRow;
+        sorted.forEach((d: Datum) => {
+            const start = +d[startField] - paddingInBp;
+            const end = +d[endField] + paddingInBp;
```

Comment:
These are the lines that solve the issue. The `d[endField]` was not converted to `number` previously. It was `const end = (d[endField] as number) + paddingInBp;` and `end` was a string value.